pkg/interface/chat/src/js/lib/util.js

```diff
@@ -67,7 +67,30 @@ export function uxToHex(ux) {
     let value = ux.substr(2).replace('.', '').padStart(6, '0');
     return value;
   }
-  
+
   let value = ux.replace('.', '').padStart(6, '0');
   return value;
 }
+
+export function writeText(str) {
+  return new Promise(function (resolve, reject) {
+
+    var range = document.createRange();
+    range.selectNodeContents(document.body);
+    document.getSelection().addRange(range);
+
+    var success = false;
+    function listener(e) {
+      e.clipboardData.setData("text/plain", str);
+      e.preventDefault();
+      success = true;
+    }
+    document.addEventListener("copy", listener);
+    document.execCommand("copy");
```

Comment:
I don't buy that this works every time. It relies on a race condition of the document.addEventListener being executed before we get to line 94.

Reply:
These code blocks are asynchronous...

Reply:
I got this from a [gist for a Safari-compatible copy](https://gist.github.com/lgarron/d1dee380f4ed9d825ca7#gistcomment-2934251). I gotta read through this again (it’s just older JS that’s bleh to look at) and see what I can do here.